activesupport/lib/active_support/cache.rb

```diff
@@ -478,7 +478,7 @@ def fetch(name, options = nil, &block)
           if entry
             get_entry_value(entry, name, options)
           else
-            save_block_result_to_cache(name, options, &block)
+            save_block_result_to_cache(name, options.dup, &block)
```

Comment:
I think it would makes sense to push this down into `save_block_result_to_cache`, since that's where the options are passed to the block:

```diff
index ab4deb9b97..79c0c19b39 100644
--- a/activesupport/lib/active_support/cache.rb
+++ b/activesupport/lib/active_support/cache.rb
@@ -478,7 +478,7 @@ def fetch(name, options = nil, &block)
           if entry
             get_entry_value(entry, name, options)
           else
-            save_block_result_to_cache(name, options.dup, &block)
+            save_block_result_to_cache(name, options, &block)
           end
         elsif options && options[:force]
           raise ArgumentError, "Missing block: Calling `Cache#fetch` with `force: true` requires a block."
@@ -1053,6 +1053,8 @@ def get_entry_value(entry, name, options)
         end

         def save_block_result_to_cache(name, options)
+          options = options.dup
+
           result = instrument(:generate, name, options) do
             yield(name, WriteOptions.new(options))
           end
```

What do you think?

Reply:
yeah, makes sense.